Sources/SWBTaskConstruction/TaskProducers/OtherTaskProducers/CustomTaskProducer.swift

```diff
@@ -22,7 +22,14 @@ final class CustomTaskProducer: PhasedTaskProducer, TaskProducer {
             for customTask in context.configuredTarget?.target.customTasks ?? [] {
                 
                 let commandLine = customTask.commandLine.map { context.settings.globalScope.evaluate($0) }
-                let environment = EnvironmentBindings(customTask.environment.map { (context.settings.globalScope.evaluate($0.0), context.settings.globalScope.evaluate($0.1)) })
+                var environmentAssignments = computeScriptEnvironment(.shellScriptPhase, scope: context.settings.globalScope, settings: context.settings, workspaceContext: context.workspaceContext)
+                if context.workspaceContext.core.hostOperatingSystem != .macOS {
```

Comment:
what's the goal here?

Reply:
Because it overwrites the system environment, or what?

Reply:
I wonder if it's worth stripping unconditionally; the main use was inspecting SDKROOT anyways, and you can get at DEVELOPER_DIR, TOOLCHAIN_DIR, etc, to find anything else presumably relevant.

Reply:
yep, in some tests we end up with a mix of different Path and PATH

Reply:
> yep, in some tests we end up with a mix of different Path and PATH

Ah, of course. We have #186 to improve that situation.